src/DataGrid.tsx

```diff
@@ -492,11 +485,11 @@ function DataGrid<R, SR, K extends Key>(
     onExpandedGroupIdsChange(newExpandedGroupIds);
   }
 
-  function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>, isEditorPortalEvent = false) {
+  function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>) {
     if (!(event.target instanceof Element)) return;
     const isCellEvent = event.target.closest('.rdg-cell') !== null;
     const isRowEvent = hasGroups && event.target.matches('.rdg-row, .rdg-header-row');
-    if (!isCellEvent && !isRowEvent && !isEditorPortalEvent) return;
```

Comment:
`isEditorPortalEvent` was added to handle `Enter` and `Escape`. I think we do need this for portal editors as portal editors would be usually complex like Modal etc